### ISSUE TITLE
[Feature Request] A Custome Search list that provides results in addons.

### ISSUE THREAD

**Issue by Goldenfreddy0703** [drinfernoo/plugin.program.autowidget#103]
**Is your feature request related to a problem? Please describe.**
This feature request is not typically a problem but it is on some skins cause most skins don't prove a costume search engine to edit with.

**Describe the solution you'd like**
A  Costume Search List that provides results in addons.
Ok so basically, i think its possible to make a Costume Search List so that what it does is that it searches in costume add-ons for results so if you wanted to search in every add-on, you can with a costume search list. 

Here is a link that shows what i mean:
jurialmunkey/skin.arctic.horizon.2#104

Example:
![image](https://user-images.githubusercontent.com/62456796/128564652-2290a812-26e3-4e35-b273-4bdaa1dfd9c3.png)
![image](https://user-images.githubusercontent.com/62456796/128565451-95ac4d13-4d5c-43b3-a3ae-fcbdf463d226.png)

You would provide the action and the aspect of it, and when you want to search something, we could use your addon and find the search results in all these other addons. It can similar to the SALTS addon back then in which this could be the first addon for adding a search in engine in any skin you like. 

**Describe alternatives you've considered**
Using Heavy Weight Auramod or Artic Horizon sense they have search list. 

Only downside: Heavy Weight 

**Additional context**
The reason why i made this a Feature Request in your addon is cause your addon is perfect and it helped me alot especially with seren when there was no icons to begin with. Your addon is the great icon and widget tool ever and is way better then super favorites, having a custom engine would be an extra useful tool that would help every skin people use.

**Comment by drinfernoo** [comment]
I don't really see how this would work in AutoWidget... The "custom search paths" feature in those skins allows you to have widgets in a "search hub" that each show separate results from a specific add-on.

The closest I could get would be to have paths in AutoWidget that you could then point homescreen widgets at... but how would you get the search query into them? And how would you get them into that "search hub"?

**Comment by Goldenfreddy0703** [comment]
You know, i just realized and im not too sure about that. I thought about this after making this suggestion and i think when it comes to search results, you ask yourself how its going to be displayed and im not too sure about that. The only solution to that is probably making a costume menu in order to show search results from the addons you made the custom search paths on. I thought it would be a good idea but now im thinking it might not due to the multiple results, You would probably have to create a custom menu for that as well in order to get multiple results. 

Now i kinda wonder how the SALTS addon worked with the search and stuff, it would search through addons but idk how it displayed the results.

If you want to talk more about this Feature Request. My discord is The Owl Witch#0703

**Comment by drinfernoo** [comment]
I just don't really think that this is something that's within the scope of what AutoWidget is designed to do.

**Comment by djay** [comment]
@drinfernoo slightly different feature but in theory there could be a search titles of cached paths. not sure how useful that would be though.

**Comment by djay** [comment]
@drinfernoo actually there is perhaps a way put search into autowidget. Someone could do a search in another plugin and then click "add to autowidget", then it asks them for the keyword they searched for so it knows what to replace in the url.
Then autowidget could have a search group of multiple search paths which can be merged. Browsing the search group asks for a keyword and does the search. Then skins could use use autowidgets search of that group to combine search results.

If autowidget is about ways to combine multiple addon lists togeather then I can see how it could be within scope?

**Comment by Goldenfreddy0703** [comment]
@djay hey I was thinking about that too but having autowidge combining search results is probably not as useful as much. You did give me a good idea tho. Every 15 sec or 20 sec, the search results could cycle through so maybe it is possible to have your own search paths and would be a good idea.

What do you think @drinfernoo

**Comment by djay** [comment]
I think it would only make sense if each plugin results list was very short, ie a very specific search.
or otherwise there would need to be a relevence ranking alg to work out how to combine results from plugins so better results are first. and/or remove duplicates of the same item if the aim was that different plugins searched different databases but some results are the same.
But for example if you had a bunch of local channel providers like BBC, NBC etc and could search across them to find a show? Say an olympics event replay?

**Comment by drinfernoo** [comment]
But if the skin doesn't allow custom search paths, how will a "search" AutoWidget be supplied with the query to replace in its own paths?

**Comment by Goldenfreddy0703** [comment]
Well what you can do is make a search button and you can have settings for it to add custom search paths or actions and when you click the search button it will show you the results but can cycle through every 15 seconds.

**Comment by Goldenfreddy0703** [comment]
@djay your probably better at explaining this then me tbh.

**Comment by drinfernoo** [comment]
But where would this search button *be*? The only places AutoWidget would be **able** to put it would be either inside the add-on (in a menu), or as an item in the widget.

**Comment by Goldenfreddy0703** [comment]
You could put the search button inside of the addon so that when we customize are skins menu that search button can be used as an action.

**Comment by drinfernoo** [comment]
I've been thinking about it and I think I actually have a pretty cool way of implementing this that could be really useful. I'll reopen the issue, that way I don't forget about it :+1:

**Comment by Goldenfreddy0703** [comment]
Nice, I'm so happy to see this issue reopened and maybe it can be possible in the future. I will be glad to help you all the way on this enhancement, thank you

**Comment by djay** [comment]
> But where would this search button _be_? The only places AutoWidget would be **able** to put it would be either inside the add-on (in a menu), or as an item in the widget.

I thought the whole idea was to make it so it integrates with skin custom search paths. so it just needs to make available a path without the search filled in if I read the original request correctly. 

I guess it could also be made available to use within AW menus itself, maybe under active widgets? or special menu?

**Comment by drinfernoo** [comment]
I think his idea was to use this in skins that don't have that feature.

**Comment by Goldenfreddy0703** [comment]
Yes that was my idea sense most skins like osmc or titan bengie or embury skin Limit you on certain things especially when it comes the view types.

**Comment by drinfernoo** [comment]
There's still some bugs to work out but I have a "proof of concept" for this which I think will work pretty well 😉

**Comment by Goldenfreddy0703** [comment]
Nice, thank you so much for considering this idea. Your addon is the best and it helped me alot especially with seren.

**Comment by Goldenfreddy0703** [comment]
Hey there, i know its been a few months and its been awhile but i wanted to go back to the feature request and talk about another way we can use the search future in the auto widget. 

Another way of making a custom search is by making a custom shortcut that can produce results by typing costume search paths: jurialmunkey/skin.arctic.horizon.2#104

If we can put custom search paths into your addon and then make it as a custom shortcut then we can produce results in that shortcut. 

If it is hard to understand then i can explain it more in discord: The Owl Witch#4432

**Comment by drinfernoo** [comment]
I haven't forgotten about this, just been very busy with other things.

**Comment by Goldenfreddy0703** [comment]
Oh then that's totally fine and I understand, thank you.
